atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/SnapshotTransaction.java

```diff
@@ -747,13 +750,17 @@ private static boolean isSingleton(Iterable<?> elements) {
         };
     }
 
-    private void validateExternalAndCommitLocksIfNecessary(TableReference tableRef, long timestamp) {
-        if (!isValidationNecessary(tableRef)) {
+    private void validateExternalAndCommitLocksOnReadIfNecessary(TableReference tableRef, long timestamp) {
```

Comment:
nit: can we rename this to `validatePreCommitRequirementsOnReadIfNecessary`? Mainly because this checks the pre-commit conditions too, which may not be 'external locks'